src/Features/VisualBasic/Portable/Completion/CompletionProviders/PartialTypeCompletionProvider.vb

```diff
@@ -1,150 +1,68 @@
-
-                Case SyntaxKind.ModuleKeyword
-                    Return symbol.TypeKind = TypeKind.Module
-
-                Case SyntaxKind.InterfaceKeyword
-                    Return symbol.TypeKind = TypeKind.Interface
-
-                Case Else
-                    Return False
-
-            End Select
-
+        Protected Overrides Async Function CreateSyntaxContextAsync(document As Document, semanticModel As SemanticModel, position As Integer, cancellationToken As CancellationToken) As Task(Of AbstractSyntaxContext)
+            Return Await VisualBasicSyntaxContext.CreateContextAsync(document.Project.Solution.Workspace, semanticModel, position, cancellationToken).ConfigureAwait(False)
         End Function
 
-        Private Function InSameProject(symbol As INamedTypeSymbol, compilation As Compilation) As Boolean
-            Return symbol.DeclaringSyntaxReferences.Any(Function(r) compilation.SyntaxTrees.Contains(r.SyntaxTree))
+        Protected Overrides Function GetDisplayAndInsertionText(symbol As INamedTypeSymbol, context As AbstractSyntaxContext) As ValueTuple(Of String, String)
+            Dim displayText = symbol.ToMinimalDisplayString(context.SemanticModel, context.Position, format:=_displayTextFormat)
+            Dim insertionText = symbol.ToMinimalDisplayString(context.SemanticModel, context.Position, format:=_insertionTextFormatWithParameters)
+            Return ValueTuple.Create(displayText, insertionText)
         End Function
 
-        Private Function CreateCompletionItem(symbol As INamedTypeSymbol,
-                                              textSpan As TextSpan,
-                                              displayService As ISymbolDisplayService,
-                                              position As Integer,
-                                              context As VisualBasicSyntaxContext) As CompletionItem
-            Dim displayText As String = Nothing
-            Dim insertionText As String = Nothing
+        Public Overrides Async Function GetTextChangeAsync(document As Document, selectedItem As CompletionItem, ch As Char?, cancellationToken As CancellationToken) As Task(Of TextChange?)
+            Dim insertionText = SymbolCompletionItem.GetInsertionText(selectedItem)
 
-            If symbol.MatchesKind(SymbolKind.NamedType) AndAlso symbol.GetArity() > 0 Then
-                displayText = symbol.ToMinimalDisplayString(context.SemanticModel, position, format:=_partialNameFormat)
-                insertionText = displayText
-            Else
-                Dim displayAndInsertionText = CompletionUtilities.GetDisplayAndInsertionText(symbol, isAttributeNameContext:=False, isAfterDot:=False, isWithinAsyncMethod:=False, syntaxFacts:=context.GetLanguageService(Of ISyntaxFactsService))
-                displayText = displayAndInsertionText.Item1
-                insertionText = displayAndInsertionText.Item2
+            If ch = "("c Then
```

Comment:
It seems using `AbstractSymbolCompletionProvider` would also change the commit-on-space behaviour so, one way or another, we're going to end up special-casing something here.